cmd/root.go

```diff
@@ -63,14 +63,24 @@ A mutating webhook for Kubernetes, pointing the images to a new location.`,
 		//metricsRec := metrics.NewPrometheus(promReg)
 		log.Trace().Interface("config", cfg).Msg("config")
 
-		rClient, err := registry.NewECRClient(cfg.Target.AWS.Region, cfg.Target.AWS.EcrDomain(), cfg.Target.AWS.AccountID, cfg.Target.AWS.Role, cfg.Target.AWS.ECROptions.AccessPolicy, cfg.Target.AWS.ECROptions.LifecyclePolicy)
+		// Create registry clients for source registries
+		sourceRegistryClients := []registry.Client{}
+		for _, reg := range cfg.Source.Registries {
+			sourceRegistryClient, err := registry.NewClient(reg)
+			if err != nil {
+				log.Err(err).Msg(fmt.Sprintf("error connecting to source registry at %s", reg.GetServerAddress()))
+				os.Exit(1)
```

Comment:
I wonder if this is being too aggressive. If there are multiple sources and one fails it stops everything? Is that your expected failure mode?

Reply:
In our use case, properly connecting to source registries is mandatory; while writing this we had in mind that if we pass registries in the configuration, they should work and that if they don't, then it's preferable to abort everything rather than to start, thinking we passed some configuration which actually didn't apply because of the initial connection failure.
However I agree that it should not always be mandatory, maybe we could add a mechanism to retry the connection in the background in case of error, so image-swapper can start as usual in the meantime and eventually match its configuration as the connection to the registry.ies is restored and added back to the list of authenticated registries. 
What do you think?

Reply:
I can see the advantage of failing early. I don't think we necessarily need a retry mechanism here given the pod will be restarted by Kubernetes anyways. If this becomes an issue we can address it separately.